round-based/src/state_machine/mod.rs

```diff
@@ -25,11 +27,41 @@
 //! # todo!()
 //! }
 //!
-//! let state_machine = round_based::state_machine::wrap_protocol(
+//! // `state` implements `round_based::state_machine::StateMachine` trait.
+//! // Its methods can be used to advance protocol until completion.
+//! let mut state = round_based::state_machine::wrap_protocol(
 //!     |party| protocol_of_random_generation(party, 0, 3)
 //! );
-//! // `state_machine` implements `round_based::state_machine::StateMachine` trait.
-//! // Its methods can be used to advance protocol until completion.
+//!
+//! // Note: this is just an example. If you have stream/sink, you don't probably need to
+//! // use the sync API
+//! use futures::{Sink, Stream, SinkExt, StreamExt};
+//! async fn connect() -> Result<(
+//!     impl Stream<Item = anyhow::Result<round_based::Incoming<Msg>>>,
+//!     impl Sink<round_based::Outgoing<Msg>, Error = Error>
+//! )> {
+//!     // ...
+//!     # Ok((futures_util::stream::pending(), futures_util::sink::drain().sink_map_err(|err| match err {})))
+//! }
+//! let (mut incomings, mut outgoings) = connect().await?;
```

Comment:
Maybe replace the async containers with simple sync functions `send(msg)` and `recv() -> msg`?

Reply:
Yeah good idea

Reply:
Done